pyfar/dsp/dsp.py

```diff
@@ -1506,6 +1506,11 @@ def deconvolve(system_output, system_input, fft_length=None, **kwargs):
     if not system_output.sampling_rate == system_input.sampling_rate:
         raise ValueError("The two signals have different sampling rates!")
 
+    if freq_range is not None:
+        assert hasattr(freq_range, '__iter__') and len(freq_range) == 2
+    else:
```

Comment:
We don't need to do the check here. It is done in the regularized inversion.
```suggestion
    if freq_range is None:
```

Reply:
I added it to prevent unecessary computations. I guess the zero padding is not that bad :)